src/Neo.VM/Types/CompoundType.cs

```diff
@@ -72,8 +72,11 @@ public override int GetHashCode()
             var h = new HashCode();
             h.Add(Count);
             h.Add(Type);
-            //foreach (var item in SubItems)
-            //    h.Add(item.GetHashCode());
+            foreach (var item in SubItems)
+            {
+                if (ReferenceEquals(this, item)) continue;
```

Comment:
Not solved, if the array in inside another array

Reply:
I changed it to ignore compoundTypes.